### ISSUE TITLE
FriendsHandle.GetFriendsCount returns 0 friends

### ISSUE THREAD

**Issue by FREEZX** [PlayEveryWare/eos_plugin_for_unity#776]
**Describe the bug**
Using the provided `AuthAndFriends` sample and testing in my own app, refreshing doesn't seem to be finding any friends. Going deeper it seems like the `FriendsHandle.GetFriendsCount` function returns 0 as the number of friends.

I can see the correct user id in the logs, it's the same as the one I can see in my epic app, so it shouldn't be an issue with the account I'm using.

I also tried this in a build, where the overlay correctly opens and displays my friends while I see no friends on the right side in the app's UI. Could I be missing something with the setup, or is this a bug?

**To Reproduce**
Open the AuthAndFriends sample, log in using AccountPortal with an account that has friends.
Observe that there are no friends displayed.

**Expected behavior**
My friends are returned.

**Screenshots**
![Skopje_fyegwPSAB9](https://github.com/PlayEveryWare/eos_plugin_for_unity/assets/243900/7e89b071-2b70-486d-81d3-efb3f0619e85)
Screenshot displaying the correctly displayed overlay with the result of 0 friends in the background

**Desktop (please complete the following information):**
 - OS: Windows x64
 - Editor and standalone build
 - 2022.3.29f1
 - 3.2.1

**Comment by WispyMouse** [comment]
Ahoy @FREEZX ! Let's work on fixing this and freeing up your friends.

I have discovered something new to me in working on investigating this. It turns out that `QueryFriends` can only find friends who have consented to this application to read their online information. Generally that happens when someone logs in to the game for the first time, there'll be a popup asking for consent. Here's a demonstration of that.

My Epic Account has several friends. One of them has authorized this account before, so the "Friends" tab shows them.

![image](https://github.com/PlayEveryWare/eos_plugin_for_unity/assets/16852552/be702bea-ca24-4715-919f-33488504769e)

If I open the Social Overlay, that has extra special permission to see friends, so my friends show up.

![image](https://github.com/PlayEveryWare/eos_plugin_for_unity/assets/16852552/70e4ba5d-7106-4c51-9757-db931451a227)

It's pretty un-intuitive that it works this way! [The Documentation on the Friends Interface](https://dev.epicgames.com/docs/epic-account-services/eos-friends-interface#retrieving-and-caching-the-friends-list) mentions this.

> Friends Interface functions require this handle to access your signed-in user's friends list. It is important to note that [EOS_Friends_QueryFriends](https://dev.epicgames.com/docs/en-US/api-ref/functions/eos-friends-query-friends) will only return friends who have played the game at least once and have given their consent. Consent is either give upon game purchase or when a player signs in to an application using Epic Account Services (EAS) and allows EAS access to their account data.

Our plugin's `EOSFriendsManager` doesn't mention this at all, so I've created a task in our back end task tracking system to document this more clearly.
